programs/clearing_house/src/math/margin.rs

```diff
@@ -341,6 +364,75 @@ pub fn calculate_margin_requirement_and_total_collateral(
     Ok((margin_requirement, total_collateral))
 }
 
+pub struct TokenValueMap(pub BTreeMap<u64, (i128, i128)>);
+
+pub fn build_token_value_map(
+    user: &User,
+    bank_map: &BankMap,
+    oracle_map: &mut OracleMap,
+) -> ClearingHouseResult<TokenValueMap> {
+    let mut token_value_map: TokenValueMap = TokenValueMap(BTreeMap::new());
+    token_value_map.0.insert(0, (0, 0));
+    for user_bank_balance in user.bank_balances.iter() {
+        if user_bank_balance.balance == 0 || user_bank_balance.open_orders == 0 {
+            continue;
+        }
+
+        let bank = bank_map.get_ref(&user_bank_balance.bank_index)?;
+        if user_bank_balance.bank_index == 0 {
+            let quote_token_amount = get_signed_token_amount(
+                get_token_amount(
+                    user_bank_balance.balance,
+                    &bank,
+                    &user_bank_balance.balance_type,
+                )?,
+                &user_bank_balance.balance_type,
+            )?;
+
+            let (quote_token_amount_before, quote_token_value_before) =
+                token_value_map.0.get(&0).unwrap();
+            token_value_map.0.insert(
+                0,
+                (
+                    quote_token_amount_before
+                        .checked_add(quote_token_amount)
+                        .ok_or_else(math_error!())?,
+                    quote_token_value_before
+                        .checked_add(quote_token_amount)
+                        .ok_or_else(math_error!())?,
+                ),
+            );
+        } else {
+            let oracle_price_data = oracle_map.get_price_data(&bank.oracle)?;
+            let (worst_case_token_amount, worse_case_quote_token_amount) =
+                user_bank_balance.get_worst_case_token_amounts(&bank, oracle_price_data)?;
+            let worst_case_token_value =
+                get_token_value(worst_case_token_amount, bank.decimals, oracle_price_data)?;
+
+            token_value_map.0.insert(
+                user_bank_balance.bank_index,
+                (worst_case_token_amount, worst_case_token_value),
```

Comment:
guess bigger q is:

does a single sell limit order for ETH -> USDC, mean the worst case here will be that it fills? doesnt that increase your margin for trading because ETH asset weight < USDC asset weight?